### ISSUE TITLE
How save checkpoint for  Resuming Training

### ISSUE THREAD

**Issue by remotejob** [albertlai431/Machine-Learning#1]
simple

 `model_name = 'model1.pt'

checkpoint = {'n_hidden': net.n_hidden,
              'n_layers': net.n_layers,
              'state_dict': net.state_dict(),
              'tokens': net.chars}

with open(model_name, 'wb') as f:
    torch.save(checkpoint, f)
 
`

looks like not enough to resuming training
by:

`checkpoint = torch.load(model_name)
net = CharRNN(checkpoint['tokens'])
net.load_state_dict(checkpoint['state_dict'])
`  
My question: What else need to be save?
A part of n_hidden n_layers tokens
Optimazer? Loss?
